spec/v1.0-draft/field_types.md

```diff
@@ -46,6 +46,50 @@ _DigestSet (object)_
 > -   `{"sha256": "abcd"}` does not match `{"sha256": "fedb", "sha512": "abcd"}`
 > -   `{"somecoolhash": "abcd"}` uses a non-predefined algorithm
 
+<a id="Reference"></a>
+_Reference (object)_
+
+> A size-efficient representation of any hashable artifact referenced by
+> another in-toto attestation.
+>
+> Schema:
+>
+>```jsonc
+> {
+>   "localName": "<NAME>",
+>   "digest": { "<ALGORITHM>": "<HEX VALUE>", ... },
+>   "downloadLocation": "<RESOURCE URI>", // optional
+>   "mimeType": "<MIME TYPE>" // optional
+> }
+>```
+>
+> Fields:
+>
+> -   `localName` (_string, required_): Human-readable identifier to
+>     distinguish the referenced artifact locally. The semantics are up to
+>     the producer and consumer.
+> -   `digest` (_[DigestSet], required_): The producer and consumer must
+>     agree on acceptable algorithms.
+> -   `downloadLocation` (_[ResourceURI], optional_): The location of the
+>     referenced artifact. To enable automated downloads by consumers, the
+>     specified location SHOULD be resolvable.
+> -   `mimeType` (_string, optional_): The [MIME Type][] (i.e., media type)
+>     of the referenced artifact.
+>
+> Note: A reference is bound to the `digest` field. All other fields are
+> informational.
```

Comment:
I think it depends on what we want the semantics of a `Reference` to be. If we want to be able to point to artifacts _and_ resource-type things, I'm fine with making that possible through a URI field as long as we make the semantics and documentation clear. My other question is, is there any other information about a resource that is necessary/useful when referencing it?

Reply:
I suggest using different types, though having the same field names is probably useful to reduce confusion (i.e. `subject`'s type is a subset of a reference typ).

Reply:
To add, the `mimeType`  or `mediaType` is only useful for describing artifacts, for instance, and as you describe below, resources have potentially other properties that would be helpful for the consumer/user of a reference. I'm inclined to make a clearer distinction between references to hashable artifacts vs resources because the meaning and usage of the reference does differ between the two.

Reply:
(Just saw your latest comment, Mark, as I was in the middle of typing another comment). We've decided that the `subject` field will stay intact, but I agree that we should be as consistent as we can with the field names across of reference types. By "two types", though I was referring to non-subject references, sorry if that wasn't clear.